### PR TITLE
Implement frontend e2e link checking into new PR workflow

### LINKED ISSUE #14897: Implement frontend e2e link checking into new PR workflow
With the new PR workflow for frontend merged, etan's link checking steps need to be implemented into that flow. Some changes will need to be made to work with the sharding being done.

# Acceptance Criteria
- [ ] Link checking runs as expected for e2e tests
- [ ] Link warning log is generated and captured in shard e2e-data artifacts
- [ ] Link warning log is properly extracted from artifacts and parsed
- [ ] PR commenting logic (if a warning was generated) runs as expected

### PR DESCRIPTION
**Pull request by jpandersen87**
Fixes #14897

This PR implements the link checking logic in the `build-frontend` github action into the new `frontend-ci` workflow. Environment variable checking within the e2e spec file were moved to the `test.ts` file, and two new fixture properties were made that are read in the link checking test in order to determine if it should log and where. The bash script was modified to only create a `WARNING_BODY` env var, with comment steps split out and using actions to handle find/replace.

### CODE REVIEW

**Comment by etanb** [comment]
Chatted with Joseph, will wait till:

- [ ] Errors in Bash are resolved
- [ ] At least 2 sample broken links are shown in the PR


Before reviewing officially.

File: frontend-react/src/components/header/ReportStreamHeader.tsx
Comment (penny-lischer) [inline frontend-react/src/components/header/ReportStreamHeader.tsx:289]: What are these used for?

Comment (jpandersen87) [reply frontend-react/src/components/header/ReportStreamHeader.tsx:289]: fake links to trigger the bad link check to produce output, they will be removed after

**Comment by etanb** [comment]
@jpandersen87 it works!

File: frontend-react/src/components/header/ReportStreamHeader.tsx
Comment (etanb) [reply frontend-react/src/components/header/ReportStreamHeader.tsx:289]: Just need to remove these and LGTM

**Review by etanb — APPROVED** [review]
LGTM -- contingent on fake links being removed.

**Review by snesm — APPROVED** [review]
Generally we get rid of dead code (line 216-end)

**Review by penny-lischer — APPROVED** [review]
lgtm

**Pull request merged**
